activitysim/abm/models/disaggregate_accessibility.py

```diff
@@ -740,24 +758,26 @@ def compute_disaggregate_accessibility(network_los, chunk_size, trace_hh_id):
         "school_destination_size",
         "workplace_destination_size",
     ]:
-        pipeline.drop_table(tablename)
+        state.drop_table(tablename)
 
-    for ch in list(pipeline.get_rn_generator().channels.keys()):
-        pipeline.get_rn_generator().drop_channel(ch)
+    for ch in list(state.get_rn_generator().channels.keys()):
+        state.get_rn_generator().drop_channel(ch)
 
     # Drop any prematurely added traceables
-    for trace in [
-        x for x in inject.get_injectable("traceable_tables") if "proto_" not in x
-    ]:
-        tracing.deregister_traceable_table(trace)
-
-    # need to clear any premature tables that were added during the previous run
-    orca._TABLES.clear()
-    for name, func in inject._DECORATED_TABLES.items():
-        logger.debug("reinject decorated table %s" % name)
-        orca.add_table(name, func)
+    for trace in [x for x in state.tracing.traceable_tables if "proto_" not in x]:
+        state.tracing.deregister_traceable_table(trace)
+
+    # # need to clear any premature tables that were added during the previous run
+    for name in list(state.existing_table_status):
+        if name not in tables_prior:
+            state.drop_table(name)
```

Comment:
I think we can get rid of the above lines 757-761 since we are deleting them here anyways.  Also, it looks like tables_prior includes the proto_population.  I don't think we need to keep these tables in the pipeline soaking up resources.  For that matter, I don't think we need to keep any tables in the pipeline except the "logsums" that are added below.
Also, the orca comments below can be cleaned up.

Reply:
I dropped the dropping of two named tables as you suggest, and left the cleaning-up to the automatic filter in tables_prior.

I feel like cleaning up / dropping the proto_population tables would best be done as an explicit separate step, kind of like how `initialize_proto_population` is its own thing.  That will make it clearer what to do should anyone want to do something with the proto_pop after running `compute_disaggregate_accessibility`.